village/src/App.js

```diff
@@ -1,24 +1,107 @@
-import React, { Component } from 'react';
+import React, { Component } from "react";
+import axios from "axios";
+import { Route, NavLink } from "react-router-dom";
 
-import './App.css';
-import SmurfForm from './components/SmurfForm';
-import Smurfs from './components/Smurfs';
+import "./App.css";
+import SmurfForm from "./components/SmurfForm";
+import Smurfs from "./components/Smurfs";
+import SmurfDelete from "./components/SmurfDelete"
 
 class App extends Component {
   constructor(props) {
     super(props);
     this.state = {
       smurfs: [],
+      input: {
+        name: "",
+        age: "",
+        height: ""
+      },
+      deleteInput: ""
     };
   }
+
+  componentDidMount() {
+    this.getSmurfs();
+  }
+
+  getSmurfs = () => {
+    console.log("PANG")
+    axios
+      .get("http://localhost:3333/smurfs")
+      .then(res => this.setState({ smurfs: res.data }))
+      .catch(err => console.log(err));
+  };
+  
+  addSmurf = ev => {
+    ev.preventDefault();
+    axios
+    .post("http://localhost:3333/smurfs", this.state.input)
+    .then(res => this.setState({ input: { name: "", age: "", height: "" } }))
+    .then(this.getSmurfs)
+    .catch(err => {
+      console.log(err);
+    });
+  }
+
+  killSmurf = ev => {
+    ev.preventDefault();
+    const doomedSmurf = this.state.smurfs.find(el => this.state.deleteInput == el.name);
```

Comment:
I think this makes more sense as a button on each smurf, but this definitely works.